walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -83,10 +104,16 @@ private boolean hasCommonAttendees(Event event, Collection<String> people) {
       while (conflictIndex < conflicts.size() - 1
           && conflict.overlaps(conflicts.get(conflictIndex + 1))) {
         conflictIndex++;
-        conflict = conflicts.get(conflictIndex);
+        TimeRange nextOverlappingConflict = conflicts.get(conflictIndex);
 
-        // The next overlapping time range may or may not have a later end time.
-        conflictEnd = Math.max(conflictEnd, conflict.end());
+        // The remaining time ranges that overlap with the current should be compared to the one
+        // with the latest end time. Otherwise, in the case that a time range contains two
+        // non-overlapping time ranges, the loop would exit before all three time ranges are
+        // looked at because the third range does not overlap with the second.
+        if (nextOverlappingConflict.end() > conflictEnd) {
+          conflict = nextOverlappingConflict;
+          conflictEnd = nextOverlappingConflict.end();
+        }
       }
```

Comment:
You may want to extract this to a separate helper method. Could be called getNextAvailableStartTime() or perhaps there is a better way to refactor. Having a nested while loop that modifies the index of the outer for loop is a bit complicated.